packages/tendermint-rpc/src/tendermint34/adaptor/responses.ts

```diff
@@ -252,6 +234,29 @@ function decodeConsensusParams(data: RpcConsensusParams): responses.ConsensusPar
   };
 }
 
+// for block results and genesis
+interface RpcValidatorUpdate {
+  /** hex encoded */
+  readonly address: string;
+  readonly pub_key:
+    | {
+        Sum: RpcPubkey;
+      }
+    | RpcPubkey;
+  readonly power: string;
+  readonly name?: string;
+}
+
+function decodeValidatorUpdate(data: RpcValidatorUpdate): responses.Validator {
+  return {
+    pubkey: decodePubkey(
+      assertObject(data.pub_key && ("Sum" in data.pub_key ? data.pub_key.Sum : data.pub_key)),
+    ),
+    votingPower: Integer.parse(assertNotEmpty(data.power)),
+    address: fromHex(assertNotEmpty(data.address)),
+  };
+}
```

Comment:
Could you make the function public and create a unit test for this? I'm pretty sure this is not yet correct.